### DISCUSSION TITLE
PKIX path building failed: sun.security.provider.certpath.SunCertPathBuilderException: unable to find valid certification path to requested target

### DISCUSSION THREAD

**Discussion by eminkahramanar** [strimzi/strimzi-kafka-operator#6692] [General]
Hi,

I have deployed Strimzi on Minikube and trying to connect to cluster with AKHQ with attached properties but I got the following error

"PKIX path building failed: sun.security.provider.certpath.SunCertPathBuilderException: unable to find valid certification path to requested target"

Please Help

Thanks in advance
[strimzi.txt](https://github.com/strimzi/strimzi-kafka-operator/files/8495901/strimzi.txt)

Here is the properties I used to connect to cluster from AKHQ

            bootstrap.servers: "bootstrap.192.168.59.146.nip.io:443"
            security.protocol: SASL_SSL
            ssl.truststore.location: /app/truststore.jks
            ssl.truststore.password: xxxxxx
            ssl.truststore.type: JKS
            sasl.mechanism: SCRAM-SHA-512                                               
            sasl.jaas.config: org.apache.kafka.common.security.scram.ScramLoginModule required username="kafka" password="xxxx";

**Comment by scholzj** [comment]
I don't know anything about AKHQ. But that error normally suggests that you didn't correctly configured its truststore and it does not trust the Kafka broker when connecting to it using TLS.

**Reply by scholzj** [reply]
PS: Since you seem to be using Ingress -> it could also be caused by badly configured TLS passthrough in the Ingress controller.

**Reply by eminkahramanar** [reply]
I used following commands 

kubectl get secret my-cluster-cluster-ca-cert -n kafka -o jsonpath='{.data.ca.crt}' | base64 -d > ca.crt

keytool -import -trustcacerts -file ca.crt -keystore truststore.jks -storepass xxxx

and copied truststore.jks file to akhq pod

**Reply by eminkahramanar** [reply]
How do I detect if config of Ingress controller is correct or not ?

**Reply by scholzj** [reply]
The command looks good, so I guess you should check the Ingress. You can try for example with OpenSSL. Something like:

```
openssl s_client -connect bootstrap.192.168.59.146.nip.io:443 -servername bootstrap.192.168.59.146.nip.io -showcerts
```

**Reply by eminkahramanar** [reply]
This gives me  Kubernetes Ingress Controller Fake Certificate

**Reply by scholzj** [reply]
Ok, that means that the TLS Passthrough is not enabled / does not work.

**Reply by eminkahramanar** [reply]
Do you know how I can enable ? I used command "minikube addon enable ingress" but this doesn't enable it by default I guess

**Reply by scholzj** [reply]
There is an options for it: https://kubernetes.github.io/ingress-nginx/user-guide/tls/#ssl-passthrough ... but I do not know how to configure it for Minikube addons.
